op-deployer/pkg/deployer/intentbuilder/builder.go

```diff
@@ -0,0 +1,385 @@
+}
+
+func (c *l2Configurator) WithBaseFeeVaultRecipient(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].BaseFeeVaultRecipient = address
+}
+
+func (c *l2Configurator) WithSequencerFeeVaultRecipient(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].SequencerFeeVaultRecipient = address
+}
+
+func (c *l2Configurator) WithL1FeeVaultRecipient(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].L1FeeVaultRecipient = address
+}
+
+func (c *l2Configurator) WithL1ProxyAdminOwner(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].Roles.L1ProxyAdminOwner = address
+}
+
+func (c *l2Configurator) WithL2ProxyAdminOwner(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].Roles.L2ProxyAdminOwner = address
+}
+
+func (c *l2Configurator) WithSystemConfigOwner(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].Roles.SystemConfigOwner = address
+}
+
+func (c *l2Configurator) WithUnsafeBlockSigner(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].Roles.UnsafeBlockSigner = address
+}
+
+func (c *l2Configurator) WithBatcher(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].Roles.Batcher = address
+}
+
+func (c *l2Configurator) WithProposer(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].Roles.Proposer = address
+}
+
+func (c *l2Configurator) WithChallenger(address common.Address) {
+	c.builder.intent.Chains[c.chainIndex].Roles.Challenger = address
+}
+
+func (c *l2Configurator) WithEIP1559DenominatorCanyon(value uint64) {
+	c.builder.intent.Chains[c.chainIndex].Eip1559DenominatorCanyon = value
+}
+
+func (c *l2Configurator) WithEIP1559Denominator(value uint64) {
+	c.builder.intent.Chains[c.chainIndex].Eip1559Denominator = value
+}
+
+func (c *l2Configurator) WithEIP1559Elasticity(value uint64) {
+	c.builder.intent.Chains[c.chainIndex].Eip1559Elasticity = value
+}
+
+func (c *l2Configurator) WithOperatorFeeScalar(value uint64) {
+	c.builder.intent.Chains[c.chainIndex].OperatorFeeScalar = uint32(value)
+}
+
+func (c *l2Configurator) WithOperatorFeeConstant(value uint64) {
+	c.builder.intent.Chains[c.chainIndex].OperatorFeeConstant = value
+}
+
+func (c *l2Configurator) WithRegolithTimeOffset(offset *int) {
+	c.setHardforkTimeOffsets("Regolith", offset)
+}
+
+func (c *l2Configurator) WithCanyonTimeOffset(offset *int) {
+	c.setHardforkTimeOffsets("Canyon", offset)
+}
+
+func (c *l2Configurator) WithDeltaTimeOffset(offset *int) {
+	c.setHardforkTimeOffsets("Delta", offset)
+}
+
+func (c *l2Configurator) WithEcotoneTimeOffset(offset *int) {
+	c.setHardforkTimeOffsets("Ecotone", offset)
+}
+
+func (c *l2Configurator) WithFjordTimeOffset(offset *int) {
+	c.setHardforkTimeOffsets("Fjord", offset)
+}
+
+func (c *l2Configurator) WithGraniteTimeOffset(offset *int) {
+	c.setHardforkTimeOffsets("Granite", offset)
+}
+
+func (c *l2Configurator) WithHoloceneTimeOffset(offset *int) {
```

Comment:
While I do like having typing that separates the hardforks from the individual intent attributes and their internal relations, I think we may want to parametrize the roles and hardforks, to have a little more flexibility as caller. E.g. a test may be parametrized, and run for any of the forks. Being able to pass that fork as a typed argument, rather than having a switch-statement and calling the right function, would be neat.

Maybe fork-names and role names are the first things we can centralize in the monorepo, as typed enum-like identifiers, so tools/interfaces can all build against those without doing translation work between packages? (And `devkeys` does have some form of role identifier already, that would be nice to consolidate).

Reply:
Sure, I can centralize the fork names!

Reply:
I used the ones in the `rollup` package to avoid bloating this PR too much.